test/util/util.go

```diff
@@ -770,3 +775,15 @@ readCh:
 	}
 	gomega.ExpectWithOffset(1, gotObjs).To(gomega.Equal(objs))
 }
+
+func NewTestingLogger(writer io.Writer, level int) logr.Logger {
```

Comment:
Can we move this function to https://github.com/kubernetes-sigs/kueue/tree/main/pkg/util/testing?
It seems that this has a similar responsibility as context and client.

Reply:
it can get confusing, in `pkg/util/testing/context.go` is more targeted at unit tests (using a testing.T based logger), this logger is a "standard" logger and used in integration and e2e only.

Reply:
Uhm, that sounds reasonable.
/lgtm
/approve